gcapi/gcapi.py

```diff
@@ -539,13 +537,47 @@ def __call__(
         else:
             return response
 
+    def run_external_job(
+        self, *, algorithm_name: str = None, inputs: Dict[str, any]
+    ):
+        algorithm = self._get_algorithm(algorithm_name=algorithm_name)
+        input_interfaces = {ci["title"]: ci for ci in algorithm["inputs"]}
+
+        for ci in input_interfaces:
+            if ci not in inputs and input_interfaces[ci]["default_value"] is None:
+                raise ValueError(f"{ci} is not provided")
+
+        job = {"algorithm_slug": algorithm["slug"], "inputs": []}
```

Comment:
Should we send `algorithm_slug`, or change the serializer on the gc side an call it `algorithm` and then use a pk or hyperlinked related field?

Reply:
Ok, changed it to HyperlinkedRelatedField.